addons/binding/org.openhab.binding.kvv/src/main/java/org/openhab/binding/kvv/internal/KVVHandler.java

```diff
@@ -0,0 +1,226 @@
+
+        for (final Channel channel : thing.getChannels()) {
+            this.channels.add(channel.getUID());
+            logger.info(channel.getUID().getAsString());
+        }
+
+        for (final String key : thing.getConfiguration().getProperties().keySet()) {
+            logger.info(key + " -> " + thing.getConfiguration().get(key));
+        }
+
+        this.config = getConfigAs(KVVConfiguration.class);
+        this.departures = null;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        this.handleRefresh(channelUID);
+    }
+
+    /**
+     * Refreshes all {@link Channel Channels}
+     */
+    public void refreshAll() {
+        for (final ChannelUID channel : this.channels) {
+            this.handleRefresh(channel);
+        }
+    }
+
+    /**
+     * Refreshes a specific {@link Channel}
+     *
+     * @param channelUID the {@link ChannelUID} of the {@link Channel}
+     */
+    @SuppressWarnings("null")
+    private void handleRefresh(ChannelUID channelUID) {
+        final int id = Integer.parseInt(channelUID.getId().replaceAll("[^0-9]", ""));
+        final String type = channelUID.getId().substring(channelUID.getId().lastIndexOf('_') + 1);
+
+        logger.debug("Refresh for '" + channelUID.getId() + "'. Id is #" + id + ", type is '" + type + "'");
+
+        if (this.departures == null) {
+            logger.info("departure is null...");
+            updateState(channelUID, new StringType(""));
+            return;
+        }
+
+        if (this.departures.getDepartures().size() - 1 < id) {
+            logger.info("Train with #" + id + " does not exist.");
+            updateState(channelUID, new StringType(""));
+            return;
+        }
+
+        final Departure departure = this.departures.getDepartures().get(id);
+        if (departure == null) {
+            updateState(channelUID, new StringType(""));
+            return;
+        }
+
+        switch (type) {
+            case "name":
+                updateState(channelUID, new StringType(departure.getRoute()));
+                break;
+            case "destination":
+                updateState(channelUID, new StringType(departure.getDestination()));
+                break;
+            case "eta":
+                updateState(channelUID, new StringType(departure.getTime()));
+                break;
+            default:
+                updateState(channelUID, new StringType(""));
+        }
+    }
+
+    /**
+     * Updates the local list of departures.
+     *
+     * @param departures the new list of departures
+     */
+    private synchronized void setDepartures(final DepartureResult departures) {
+        this.departures = departures;
+
+    }
+
+    @Override
+    public void initialize() {
+        logger.error("Start initializing!");
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            logger.info("Starting inital fetch");
```

Comment:
Please check the log level of all logging statements. This logging (and most other logging) should be done on debug. See the [Logging Guidelines](https://www.openhab.org/docs/developer/development/guidelines.html#e-logging). 

Also don't call `.toString()` on parameter objects when logging. This is will automatically be done by the logger (but only if the configured log level is high enough). This way the relatively expensive `toString` operations are only executed when strictly necessary.